### PR TITLE
Add Missing Swingame Functionalities Documentation

### PR DESCRIPTION
**Pull request by The-0x**
# Description

Documentation on the functionalities Splashkit is missing from Swingame, an old game engine made by Swinburne University that Splashkit is built off of. This document aims to gather the collection missing functions, primarily focusing on collision logic and a splash screen procedure, to be implemented with a Splashkit animation. 

## Type of change
- [x] Documentation (new)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by DarrenSunandar** [comment]
The documentation does an excellent job of pointing out and documenting the features that Swingame missing but that Splashkit might provide, with a special emphasis on collision logic. It is easy to understand and has a clear, well-organized structure with separate sections. One of the good things is that the thorough explanations of missing functions include important details that will help with future development.

**Review by DarrenSunandar — APPROVED** [review]
The documentation does an excellent job of pointing out and documenting the features that Swingame missing but that Splashkit might provide, with a special emphasis on collision logic. It is easy to understand and has a clear, well-organized structure with separate sections. One of the good things is that the thorough explanations of missing functions include important details that will help with future development.

File: docs/Splashkit/Extensions/MissingSwingameFunctionalities.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+## Approach
+
+It is suggested that the missing functionalities are in relation to collision logic. By looking 
+into the `src` files within the 
+[Swingame repository](https://github.com/macite/swingame/tree/develop/CoreSDK/src) and comparing 
+them to that of the Splashkit `src` files, missing functionalities are found and documented.
+
+## To be added to Splashkit
+The following funtions and procedures are located within the
```
Comment (Liquidscroll) [inline docs/Splashkit/Extensions/MissingSwingameFunctionalities.md:17]: Typo here, 'funtions' should be 'functions'.

File: docs/Splashkit/Extensions/MissingSwingameFunctionalities.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+It is suggested that the missing functionalities are in relation to collision logic. By looking 
+into the `src` files within the 
+[Swingame repository](https://github.com/macite/swingame/tree/develop/CoreSDK/src) and comparing 
+them to that of the Splashkit `src` files, missing functionalities are found and documented.
+
+## To be added to Splashkit
+The following funtions and procedures are located within the 
+[src](https://github.com/macite/swingame/tree/develop/CoreSDK/src) folder of the Swingame Github 
+reposotiory, but cannot be located within the
```
Comment (Liquidscroll) [inline docs/Splashkit/Extensions/MissingSwingameFunctionalities.md:19]: Typo here, 'reposotiory' should be 'repository'

File: docs/Splashkit/Extensions/MissingSwingameFunctionalities.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+into the `src` files within the 
+[Swingame repository](https://github.com/macite/swingame/tree/develop/CoreSDK/src) and comparing 
+them to that of the Splashkit `src` files, missing functionalities are found and documented.
+
+## To be added to Splashkit
+The following funtions and procedures are located within the 
+[src](https://github.com/macite/swingame/tree/develop/CoreSDK/src) folder of the Swingame Github 
+reposotiory, but cannot be located within the 
+[Splashkit src](https://github.com/splashkit/splashkit-core/tree/develop/coresdk/src/coresdk) 
+files. All descriptions are taken from their given depending Swingame`src` file.
```
Comment (Liquidscroll) [inline docs/Splashkit/Extensions/MissingSwingameFunctionalities.md:21]: I am a little confused with this sentence? Are you trying to say something like, "All descriptions are taken from the associated SwinGame `src` file." Or something like this?

File: docs/Splashkit/Extensions/MissingSwingameFunctionalities.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+Note: Andrew Cain suggests implementing the `ShowSwinGameSplashScreen` procedure with a 
+Splashkit animation instead. 
+
+- ` procedure ShowSwinGameSplashScreen();`
+    -   Shows the SwinGame intro splash screen. It would be great if you could include 
+        this at the start of your game to help us promote the SwinGame API.
+
+## Further Contributions
+
+Contributions on more missing funtionalities from Swingame or removal of funtionalities
```
Comment (Liquidscroll) [inline docs/Splashkit/Extensions/MissingSwingameFunctionalities.md:122]: 2 typos; 'funtionalities' should be 'functionalities'.

**Review by Liquidscroll — CHANGES_REQUESTED** [review]
Overall this looks good! There are few typos that I'd like to be fixed and a sentence that I'm a little confused about and might need to be changed.

File: docs/Splashkit/Extensions/MissingSwingameFunctionalities.md (hunk 3 above)
Comment (The-0x) [reply docs/Splashkit/Extensions/MissingSwingameFunctionalities.md]: That is what I meant, just couldn't word it just right.

**Review by Liquidscroll — APPROVED** [review]
Looks good to me! This will be a good resource for us to work off as we implement these functions :)

**Review by WhyPenguins — APPROVED** [review]
Nice work, looks good! Hopefully we can start populating the backlog based on this now 😄

**Pull request merged**